network/src/metrics.rs

```diff
@@ -1,155 +1,150 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+use anemo_tower::callback::MakeCallbackHandler;
+use anemo_tower::callback::ResponseHandler;
+use prometheus::{
+    register_histogram_vec_with_registry, register_int_counter_vec_with_registry,
+    register_int_gauge_vec_with_registry, IntGaugeVec, Registry,
+};
+use prometheus::{HistogramVec, IntCounterVec};
 use std::sync::Arc;
-
-pub trait NetworkMetrics {
-    fn network_available_tasks(&self) -> &IntGaugeVec;
+use std::time::Instant;
+
+#[derive(Clone)]
+pub struct NetworkMetrics1 {
+    /// Counter of requests, route is a label (ie separate timeseries per route)
+    requests: IntCounterVec,
+    /// Request latency, route is a label
```

Comment:
Maybe harmonize the format/descriptiveness of these comments? Or just delete them? The metrics are self documenting in the descriptions below anyway, so partially duplicating those descriptions here is maybe liable to get outdated/misleading

Reply:
This is specifically rustdoc comment and always prefer more doc comments